### ISSUE TITLE
Mobx Observer: You are trying to use 'observer' on a component that already has 'inject'. Please apply 'observer' before applying inject

### ISSUE THREAD

**Issue by jsappme** [mobxjs/mobx#648]
I am getting the following warning after upgrading my yarn npm packages to the latest version:

Mobx Observer: You are trying to use 'observer' on a component that already has 'inject'. Please apply 'observer' before applying inject

my packages:
"mobx": "^2.6.2",
"mobx-react": "^3.5.8",
"react": "15.3.2",
"react-native": "^0.37.0",

my appStore:

```
import { observable, action, autorun, computed } from 'mobx'
import { Actions } from 'react-native-mobx'

class AppStore {
  @observable userid = null
  @observable username = ''
  @observable posts = []
}

const appStore = new AppStore()

autorun(() => {
  console.log(appStore)
})

export default appStore
```

and the code causing the issue:

```
import { observer } from 'mobx-react/native'
import { Actions } from 'react-native-mobx'
@observer(['appStore'])
export default class LoginScreen extends Component {
```

I have no 'inject' anywhere

**Comment by mweststrate** [comment]
@jsappme this should have been fixed in mobx-react 4.0.0. Can you confirm whether this is still an issue?

**Comment by jsappme** [comment]
yes it has been fixed and I changed my code to:

@inject("appStore") @observer
